### DISCUSSION TITLE
Running gunicorn as the web server instead of uvicorn

### DISCUSSION THREAD

**Discussion by Tobi-De** [fief-dev/fief#34] [Q&A] [answered]
Hi @frankie567, hope you are doing well. I have a quick question for you if you don't mind (it is not directly related to fief). I tried to reproduce your supervisor + Docker setup but using gunicorn instead of uvicorn as a web server, and I couldn't get the application to work, I was wondering if you had encountered the same problem as me by chance since I noticed gunicorn among the fief dependencies.

**Comment by frankie567** [comment, marked as answer]
Hi @Tobi-De!

No problem! It could help if you share your supervisord configuration and error messages if any.

That said, I must tell you that doing what I did in this Docker image is usually considered as a bad practice. The container philosophy is to have only one process per container. I went against this rule here because I wanted people to get up-and-running quickly. However, this should be considered as a transitive state: for production purposes, a proper environment with a server container, a worker container and a separate Redis container should be configured.

**Reply by Tobi-De** [reply]
Thanks @frankie567 , I think I already got what I wanted with your answer, by the way keep up the good work with fief, this project looks really promising. I haven't had time to play with it myself but I will definitely try it out.
Je viens de comprendre le sens du mot `fief`, franchement brillant.

My initial problem was related to the exposed port, I was running gunicorn on a different port than the one defined in the Dockerfile.

**Reply by frankie567** [reply]
> Je viens de comprendre le sens du mot fief, franchement brillant.

Merci 😄
